lib/interjectable.rb

```diff
@@ -19,10 +27,12 @@ module ClassMethods
     #     @dependency ||= instance_eval(&default_block)
     #   end
     def inject(dependency, &default_block)
+      raise MethodAlreadyDefined if instance_methods(false).include?(dependency)
```

Comment:
1. Can you add an error message with the method name? To make debugging easier

2. Why not use `method_defined?` here? Because we only care if we're re-defining a method in this class and not an override?

Reply:
1. Yes.
2. Yes. If you want to add a new method that can call `super`, that's fine.